impl/src/main/java/org/pentaho/platform/genericfile/providers/repository/RepositoryFileProvider.java

```diff
@@ -441,4 +457,64 @@ private EnumSet<RepositoryFilePermission> getRepositoryPermissions( EnumSet<Gene
 
     return repositoryFilePermissions;
   }
+
+  @NonNull
+  @Override
+  public List<IGenericFile> getDeletedFiles() throws OperationFailedException {
+    return fileService.doGetDeletedFiles().stream()
+      .map( fileDto -> {
+        GenericFilePath parentPath = null;
+
+        try {
+          parentPath = GenericFilePath.parseRequired( fileDto.getPath() ).getParent();
+        } catch ( InvalidPathException e ) {
+          // noop
+        }
+
+        RepositoryObject repositoryObject =
+          convertFromNativeFileDto( fileDto, parentPath != null ? parentPath.toString() : null );
+
+        repositoryObject.setOwner( getOwnerFromNativeFileDto( fileDto ) );
+        repositoryObject.setOriginalLocation( getLocation( fileDto.getOriginalParentFolderPath() ) );
+        repositoryObject.setDeletedBy( fileDto.getCreatorId() );
+        repositoryObject.setDeletedDate( getDate( fileDto.getDeletedDate() ) );
+
+        return repositoryObject;
+      } )
+      .collect( Collectors.toList() );
+  }
+
+  private List<IGenericFile> getLocation( String path ) {
+    GenericFilePath locationPath = null;
+
+    try {
+      locationPath = GenericFilePath.parseRequired( path );
+    } catch ( InvalidPathException e ) {
+      // noop
+    }
+
+    List<IGenericFile> location = new ArrayList<>();
+
+    while ( locationPath != null ) {
+      IGenericFile folder;
+
+      try {
+        folder = getFile( locationPath );
+      } catch ( OperationFailedException e ) {
+        // Location folder not found, most likely because it was deleted.
+        GenericFilePath parentPath = locationPath.getParent();
+        String name = locationPath.getSegments().get( locationPath.getSegments().size() - 1 );
```

Comment:
`String name = locationPath.getLastSegment();`

Reply:
I actually thought this existed already, given I had a local uncommitted version for one 🤣 !!